TransIT.BLL/Services/ImplementedServices/PartService.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using TransIT.BLL.DTOs;
+using TransIT.BLL.Services.Interfaces;
+using TransIT.DAL.Models.Entities;
+using TransIT.DAL.UnitOfWork;
+
+namespace TransIT.BLL.Services.ImplementedServices
+{
+    /// <summary>
+    /// Part CRUD service
+    /// </summary>
+    /// <see cref="IPartService"/>
+    public class PartService : IPartService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+
+        private readonly IMapper _mapper;
+
+        public PartService(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task<PartDTO> GetAsync(int id)
+        {
+            return _mapper.Map<PartDTO>(await _unitOfWork.PartRepository.GetByIdAsync(id));
+        }
+
+        public async Task<IEnumerable<PartDTO>> GetRangeAsync(uint offset, uint amount)
+        {
+            var entities = await _unitOfWork.PartRepository.GetRangeAsync(offset, amount);
+            return _mapper.Map<IEnumerable<PartDTO>>(entities);
+        }
+
+        public async Task<IEnumerable<PartDTO>> SearchAsync(string search)
+        {
+            var parts = await _unitOfWork.PartRepository.SearchExpressionAsync(
+                search
+                    .Split(new[] { ' ', ',', '.' }, StringSplitOptions.RemoveEmptyEntries)
+                    .Select(x => x.Trim().ToUpperInvariant())
```

Comment:
I saw this part of logic with  in a few places, so maybe it can be moved inside SearchExpressionAsync method

Reply:
`search
                    .Split(new[] { ' ', ',', '.' }, StringSplitOptions.RemoveEmptyEntries)
                    .Select(x => x.Trim().ToUpperInvariant())`

Reply:
@igolovchuk I have fixed it in my branch, and will add a pull request later, after Bohdan's branch was merged